js/bundle.go

```diff
@@ -74,40 +78,40 @@ func newBundle(
 		return nil, err
 	}
 
-	// Compile sources, both ES5 and ES6 are supported.
-	code := string(src.Data)
-	c := compiler.New(piState.Logger)
-	c.Options = compiler.Options{
-		CompatibilityMode: compatMode,
-		Strict:            true,
-		SourceMapLoader:   generateSourceMapLoader(piState.Logger, filesystems),
-	}
-	pgm, _, err := c.Compile(code, src.URL.String(), false)
-	if err != nil {
-		return nil, err
-	}
 	// Make a bundle, instantiate it into a throwaway VM to populate caches.
-	rt := goja.New()
-	bundle := Bundle{
+	bundle := &Bundle{
 		Filename:          src.URL,
-		Source:            code,
-		Program:           pgm,
-		BaseInitContext:   NewInitContext(piState.Logger, rt, c, compatMode, filesystems, loader.Dir(src.URL)),
+		Source:            src.Data,
 		Options:           options,
 		CompatibilityMode: compatMode,
-		exports:           make(map[string]goja.Callable),
+		callableExports:   make(map[string]struct{}),
+		filesystems:       filesystems,
+		pwd:               loader.Dir(src.URL),
+		logger:            piState.Logger,
 		preInitState:      piState,
```

Comment:
why are you adding the `piState.logger` here, when you literally have the `piState` right below? :sweat_smile:

Reply:
fixed in [0c8c08c](https://github.com/grafana/k6/pull/2881/commits/0c8c08cf48197b776867da35cf67523ee554f84b)